src/main/scala/no/gdl/bookapi/service/search/IndexService.scala

```diff
@@ -0,0 +1,296 @@
+      ) match {
+        case Failure(failure) => errorHandler(Failure(failure))
+        case Success(response) => Success(response.result.mappings.keys.toSeq.map(_.name))
+      }
+    }
+
+    def updateAliasTarget(oldIndexName: Option[String], newIndexName: String, languageTag: LanguageTag): Try[Any] = {
+      if (!indexExisting(newIndexName).getOrElse(false)) {
+        Failure(new IllegalArgumentException(s"No such index: $newIndexName"))
+      } else {
+        var actions = List[AliasActionDefinition](AddAliasActionDefinition(BookApiProperties.searchIndex(languageTag), newIndexName))
+        oldIndexName match {
+          case None => // Do nothing
+          case Some(oldIndex) => {
+            actions = actions :+ RemoveAliasActionDefinition(BookApiProperties.searchIndex(languageTag), oldIndex)
+          }
+        }
+        esClient.execute(
+          aliases(actions)
+        ) match {
+          case Failure(failure) => errorHandler(Failure(failure))
+          case Success(_) => Success()
+        }
+      }
+    }
+
+    def deleteSearchIndex(optIndexName: Option[String]): Try[_] = {
+      optIndexName match {
+        case None => Success(optIndexName)
+        case Some(indexName) => {
+          if (!indexExisting(indexName).getOrElse(false)) {
+            Failure(new IllegalArgumentException(s"No such index: $indexName"))
+          } else {
+            esClient.execute(
+              deleteIndex(indexName)
+            ) match {
+              case Failure(failure) => errorHandler(Failure(failure))
+              case Success(_) => Success()
+            }
+          }
+        }
+      }
+    }
+
+    def aliasTarget(languageTag: LanguageTag): Try[Option[String]] = {
+
+      esClient.execute(
+        getAliases(BookApiProperties.searchIndex(languageTag),Nil)
+      ) match {
+        case Failure(failure) => Failure(failure)
+        case Success(response) => {
+          val iterator = response.result.mappings.iterator
+          iterator.hasNext match {
+            case false => Success(None)
+            case true => Success(Some(iterator.next()._1.name))
+          }
+        }
+      }
+    }
+
+    def indexExisting(indexName: String): Try[Boolean] = {
+      esClient.execute(indexExists(indexName)) match {
+        case Failure(failure) => {
+          logger.error(failure.getLocalizedMessage)
+          Success(false)
+        }
+        case Success(response) => Success(response.result.isExists)
+      }
+    }
+
+    private def getTimestamp: String = {
+      new SimpleDateFormat("yyyyMMddHHmmss").format(Calendar.getInstance.getTime)
+    }
+  }
+
+  private def errorHandler[T](failure: Failure[T]) = {
+    failure match {
+      case Failure(e: GdlSearchException) => {
+        e.getFailure.status match {
+          case notFound: Int if notFound == 404 => {
+            logger.error(s"Index ${BookApiProperties.SearchIndex} not found. Scheduling a reindex.")
+            //scheduleIndexDocuments()
```

Comment:
The error message says that a rescheduling has happened. What is supposed to happen?